retrieval/discovery/marathon/marathon.go

```diff
@@ -203,10 +217,10 @@ type AppList struct {
 }
 
 // AppListClient defines a function that can be used to get an application list from marathon.
-type AppListClient func(client *http.Client, url string) (*AppList, error)
+type AppListClient func(client *http.Client, url, metricsListener string) (*AppList, error)
 
 // fetchApps requests a list of applications from a marathon server.
-func fetchApps(client *http.Client, url string) (*AppList, error) {
+func fetchApps(client *http.Client, url, metricsListener string) (*AppList, error) {
```

Comment:
I agree with Nick that the name listener is a bit misleading. `MetricsListener` sounds better to me.